bases/shared/entrypoint.sh

```diff
@@ -400,6 +411,70 @@ get_ips() {
     done
 }
 
+get_pod_ip() {
+    # Define your variable
+    APP_LABEL_VALUE=$1
+
+    while true; do
+        POD_INFO=$(curl -sSk -H "Authorization: Bearer $TOKEN" --cacert $CA_PATH $API_ENDPOINT/api/v1/namespaces/$NAMESPACE/pods/)
+        POD_IP=$(echo "$POD_INFO" | jq --arg app_value "$APP_LABEL_VALUE" -r '.items[] | select(.metadata.labels.app == $app_value) .status.podIP')
+    
+        if [[ -z "$POD_IP" ]]; then
+            echo "Couldn't get Pod IP address. Trying again..."
+        else
+            break
+        fi
+        sleep 10
+    done
+
+    echo "$POD_IP"
+}
+
+wait_for_pod() {
+    # Define your variable
+    APP_LABEL_VALUE=$1
+
+    while true; do
+        POD_INFO=$(curl -sSk -H "Authorization: Bearer $TOKEN" --cacert $CA_PATH $API_ENDPOINT/api/v1/namespaces/$NAMESPACE/pods/)
+        POD_PHASE=$(echo "$POD_INFO" | jq --arg app_value "$APP_LABEL_VALUE" -r '.items[] | select(.metadata.labels.app == $app_value) .status.phase')
+    
+        if [[ "$POD_PHASE" != "Running" ]]; then
+            echo "Pod not running yet. Trying again..."
+        else
+            break
+        fi
+        sleep 10
+    done
+}
+
+fork_setup() {
+    THIS_FORK=$1
+    wait_for_pod "fork1"
+    wait_for_pod "fork2"
+
+    echo "Fetching IP addresses of the two nodes..."
+    FORK1_IP=$(get_pod_ip "fork1")
+    FORK2_IP=$(get_pod_ip "fork2")
+
+    mkdir -p $AGORIC_HOME
+    rm -rf $AGORIC_HOME/*
+
+    if [ ! -f "$MAINFORK_IMAGE_URL/$THIS_FORK-config-$MAINFORK_HEIGHT.tar.gz" ]; then
+        apt install -y axel
+        axel --quiet -n 10 -o "/state/$THIS_FORK-config-$MAINFORK_HEIGHT.tar.gz" "$MAINFORK_IMAGE_URL/$THIS_FORK-config-$MAINFORK_HEIGHT.tar.gz"
+        axel --quiet -n 10 -o "/state/agoric-$MAINFORK_HEIGHT.tar.gz" "$MAINFORK_IMAGE_URL/agoric-$MAINFORK_HEIGHT.tar.gz"
+
+        tar -xzf "/state/$THIS_FORK-config-$MAINFORK_HEIGHT.tar.gz" -C $AGORIC_HOME        
+        tar -xzf "/state/agoric-$MAINFORK_HEIGHT.tar.gz" -C $AGORIC_HOME
+    fi
```

Comment:
I recommend not using `hang` in this script except as controlled by the existence of `/state/hang`.  Maybe something like one of the following?

```suggestion
        # hang to debug if chain stops for any reason instead of restarting pods forever
        date > /state/hang
    fi
```

```suggestion
        # hang the first node to debug if it stops for any reason instead of restarting pods forever
        case "$THIS_FORK in
        *[^0-9]1) date > /state/hang ;;
        esac
    fi
```